File: articles/loss-aversion-losses-matter-more-than-gains/go/loss_aversion_value.go
package main

import (
	"fmt"
	"math"
)

func prospectValue(x, lambdaLoss, alphaGain, betaLoss float64) float64 {
	if x >= 0.0 {
		return math.Pow(x, alphaGain)
	}
	return -lambdaLoss * math.Pow(-x, betaLoss)
}

func main() {
	lambdaLoss := 2.0
	alphaGain := 0.88
	betaLoss := 0.88

	mixedValue := 0.5*prospectValue(240.0, lambdaLoss, alphaGain, betaLoss) +
		0.5*prospectValue(-100.0, lambdaLoss, alphaGain, betaLoss)

	fmt.Printf("Mixed gamble prospect value: %.4f\n", mixedValue)
	if mixedValue > 0.0 {
		fmt.Println("Accept mixed gamble: yes")
	} else {
		fmt.Println("Accept mixed gamble: no")
	}
}

File: articles/loss-aversion-losses-matter-more-than-gains/cpp/loss_aversion_value.cpp
#include <cmath>
#include <iostream>

double prospect_value(double x, double lambda_loss, double alpha_gain, double beta_loss) {
    if (x >= 0.0) {
        return std::pow(x, alpha_gain);
    }
    return -lambda_loss * std::pow(-x, beta_loss);
}

int main() {
    double lambda_loss = 2.0;
    double alpha_gain = 0.88;
    double beta_loss = 0.88;

    double mixed_value = 0.5 * prospect_value(240.0, lambda_loss, alpha_gain, beta_loss)
        + 0.5 * prospect_value(-100.0, lambda_loss, alpha_gain, beta_loss);

    std::cout << "Mixed gamble prospect value: " << mixed_value << std::endl;
    std::cout << "Accept mixed gamble: " << (mixed_value > 0.0 ? "yes" : "no") << std::endl;

    return 0;
}
